testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestReadUniontype.java

```diff
@@ -268,6 +268,77 @@ public void testUnionTypeSchemaEvolution(String storageFormat)
         }
     }
 
+    @Test(groups = SMOKE)
+    public void testORCUnionToStructSchemaEvolution()
```

Comment:
Please let me know if it looks good to you.

    /**
     * When reading AVRO file, Trino needs the schema information from Hive metastore to deserialize Avro files.
     * Therefore, when an ALTER table was issued in which the hive metastore changed the schema into an incompatible format,
     * from Union to Struct or from Struct to Union in this case, Trino could not read those Avro files using the modified Hive metastore schema.
     * However, when reading ORC files, Trino does not need schema information from Hive metastore to deserialize ORC files.
     * Therefore, it can read ORC files even after changing the schema.
     */